File: YuGiOh2/Cards/C66788016.cs
using System.Linq;
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 地裂
    /// </summary>
    public class C66788016
    {
        public static int Type { get; } = (int)ChooseTargetType.None;

        public static bool CheckIfAvailable(Player player)
        {
            return player.Enemy.Field.MonsterFields.Any(c => c != null && !c.Status.FaceDown);
        }

        public static void ProcessEffect(Player player)
        {
            var min = player.Enemy.Field.MonsterFields.Select(c => c == null ? int.MaxValue : c.ATK).Min();
            for (int i = 0; i < 5; i++)
            {
                if (player.Enemy.Field.MonsterFields[i] == null || 
                    player.Enemy.Field.MonsterFields[i].Status.FaceDown ||
                    player.Enemy.Field.MonsterFields[i].ATK != min)
                    continue;

                player.Enemy.AddCardToGrave(ref player.Enemy.Field.MonsterFields[i]);
                player.Enemy.Field.MonsterFields[i] = null;
                return;
            }
        }
    }
}

File: YuGiOh2/Scripts/C66788016.lua
Type = 0;
--地裂
function checkIfAvailable(player)
    for i, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.FaceDown then
            return true;
        end
    end
    return false;
end

function processEffect(player)
    local min = 999999;
    for i, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and monster.ATK < min then
            min = monster.ATK;
        end
    end
    for i, monster in pairs(player.Enemy.Field.MonsterFields) do
        if monster ~= nil and monster.ATK == min then
            player.Enemy:AddCardToGrave(monster);
            player.Enemy:ClearMonsterField(i - 1);
        end
    end
end